code/modules/mob/dead/observer/observer.dm

```diff
@@ -77,6 +96,50 @@ var/list/image/ghost_darkness_images = list() //this is a list of images for thi
 /mob/dead/CanPass(atom/movable/mover, turf/target, height=0)
 	return 1
 
+
+/*
+ * This proc will update the icon of the ghost itself, with hair overlays, as well as the ghost image.
+ * Please call update_icon(icon_state) from now on when you want to update the icon_state of the ghost,
+ * or you might end up with hair on a sprite that's not supposed to get it.
+ * Hair will always update its dir, so if your sprite has no dirs the haircut will go all over the place.
+ * |- Ricotez
+ */
+/mob/dead/observer/proc/update_icon(new_form)
+	if(hair_image)
+		overlays -= hair_image
+		ghostimage.overlays -= hair_image
+		hair_image = null
+	if(facial_hair_image)
+		overlays -= facial_hair_image
+		ghostimage.overlays -= facial_hair_image
+		facial_hair_image = null
+
+	if(new_form)
+		icon_state = new_form
+		ghostimage.icon_state = new_form
+
+	if(icon_state == "ghost") //For now we just check for the default ghost sprite. If other sprites get updated in the future to support hair, change this to a list. |- Ricotez
```

Comment:
remove the |- Ricotez from this one, the one on the big banner comment to stay.
generally 1 "I did this" comment is allowed on bigger things like procs or frameworks.